app/domain/operations/authentication/logout_saml_user.rb

```diff
@@ -10,51 +10,37 @@ class LogoutSamlUser
 
       def call(session)
         saml_settings = yield ConstructSamlSettings.new.call
-        saml_logout_xml = yield construct_saml_request_xml(saml_settings, session)
-        logout_url = yield parse_logout_service_url(saml_settings)
-        post_saml_logout_xml(logout_url, saml_logout_xml)
+        saml_logout_url = yield construct_saml_request(saml_settings, session)
+        execute_saml_logout(saml_logout_url)
       end
 
       protected
 
-      def construct_saml_request_xml(settings, session)
+      def construct_saml_request(settings, session)
         name_id = session[:__saml_name_id]
         saml_session_index = session[:__saml_session_index]
         return Failure("No SAML NameID available") if name_id.blank?
         request = OneLogin::RubySaml::Logoutrequest.new
         settings.name_identifier_value = name_id
         settings.sessionindex = saml_session_index unless saml_session_index.blank?
-        Success(request.create_logout_request_xml_doc(settings))
-      end
-
-      def parse_logout_service_url(saml_settings)
-        logout_url = saml_settings.idp_slo_service_url
-
-        parse_result = Try do
-          URI.parse(logout_url)
-        end.or(Failure("Invalid SAML logout service URI"))
-
-        return parse_result unless parse_result.success?
-
-        parsed_url = parse_result.value!
-
-        return Failure("Empty SAML service path") if parsed_url.path.blank?
-
-        Success(parsed_url)
+        Try do
+          request.create(settings)
+        end.to_result
```

Comment:
if this fails I assume it will return nil, and I think execute_saml_logout will throw an exception, where I'm going is that I think we should protect for the case the request object cannot be created, or do we need to even log it as a invalid request ?

Reply:
If it fails the raised exception will be captured as part of the failure.  Look further up in the controller where the operation is called to see where we log the failures.